File: src/main/java/com/zippeykeys/praisebe/block/base/BlockAltarBase.java
package com.zippeykeys.praisebe.block.base;

import com.zippeykeys.praisebe.deity.Deity.Type;

import net.minecraft.block.SoundType;
import net.minecraft.block.material.Material;
import net.minecraft.creativetab.CreativeTabs;

public class BlockAltarBase extends BlockEnum<Type> {
    public BlockAltarBase(String nameIn, Material materialIn, CreativeTabs... creativeTabs) {
        super(nameIn, materialIn, Type.class, creativeTabs);
        setHardness(1.5f);
        setResistance(6000000.0F);
        setHarvestLevel("pickaxe", 1);
        setSoundType(SoundType.STONE);
    }
}

File: src/main/kotlin/com/zippeykeys/praisebe/block/base/BlockAltarBase.kt
package com.zippeykeys.praisebe.block.base

import com.zippeykeys.praisebe.block.tile.base.PBTileEntity
import com.zippeykeys.praisebe.deity.Deity.Type
import net.minecraft.block.SoundType
import net.minecraft.block.material.Material
import net.minecraft.creativetab.CreativeTabs


class BlockAltarBase(nameIn: String, materialIn: Material,
        tileClass: Class<out PBTileEntity>? = null, creativeTab: CreativeTabs? = null) :
        BlockEnum<Type>(nameIn, materialIn, Type::class.java, tileClass, "type", creativeTab) {
    init {
        setHardness(1.5f)
        setResistance(6000000.0f)
        setHarvestLevel("pickaxe", 1)
        soundType = SoundType.STONE
    }
}